sources/updater/update_metadata/src/lib.rs

```diff
@@ -320,34 +319,40 @@ impl Update {
         }
     }
 
-    pub fn update_ready(&self, seed: u32) -> bool {
-        // Has this client's wave started
-        if let Some(wave) = self.update_wave(seed) {
-            return wave.has_started();
-        }
-
-        // Or there are no waves
-        true
-    }
-
-    pub fn jitter(&self, seed: u32) -> Option<DateTime<Utc>> {
+    /// Returns whether the update is truly available. A update is said to be 'ready/available' if the wave
+    /// this host belongs to has fully passed, or if the host's position in the wave has passed.
+    /// The position of the host within the wave is determined by the seed value.
+    pub fn update_ready(&self, seed: u32, time: DateTime<Utc>) -> bool {
+        // If this host is part of some update wave
         if let Some(wave) = self.update_wave(seed) {
             if wave.has_passed() {
-                return None;
+                return true;
             }
-            let bounds = match self.update_wave(seed) {
-                Some(Wave::Initial { end }) => Some((Utc::now(), end)),
-                Some(Wave::General { start, end }) => Some((start, end)),
-                Some(Wave::Last { start: _ }) | None => None,
+            let bound = match wave {
+                // For all intent and purposes, with the way we define update wave structures,
+                // the initial wave (0th wave) never has any updates.
+                // No seed value would ever land the host in this initial wave. But if for whatever reason
+                // this host is part of the initial (0th) wave, the update is going to be available
+                Wave::Initial { .. } => None,
```

Comment:
I think this comment could use another pass. The real issue is that this is impossibly hard to understand, no matter how well you write the comment! I'm going to move on, but I was just looking at whether `Wave::Initial` should exist at all. And if it doesn't serve any purpose, then I wonder what is the purpose of the seed value in the first 'wave' listed in the manifest?

Anyway for this PR, I would see if this comment could be tightened up, and maybe add doc comments to the Wave enum.

```rust
                // For all intent and purposes, with the way we define update wave structures,
                // the initial wave (0th wave) never has any updates.
                // No seed value would ever land the host in this initial wave. But if for whatever reason
                // this host is part of the initial (0th) wave, the update is going to be available
                Wave::Initial { .. } => None,
```

Reply:
This comment block might help with understanding the `Initial` wave:
https://github.com/bottlerocket-os/bottlerocket/blob/55aee0af4a0b8f7df80e58a9d49165d00d4912ff/sources/updater/update_metadata/src/lib.rs#L232-L254

I agree it's something we should look at removing (will probably have to completely restructure the wave representation in the manifest if we do). But I don't think it's appropriate for this set of changes.

Reply:
So it becomes this in `Update`?

```{
    0 => Wave::Initial{ end: <UTC datetime of 1 hour from now> },
   20 => Wave::General( start: <UTC datetime of 1 hour from now>, end: <UTC datetime of 1 day from now> ),
}
```

I didn't see that comment, but maybe those lines 241-245 should be more explicit about what the resulting map is. It feels like something isn't right about the design/implementation, but I'm not quite sure. (And if there is, fixing/simplifying may be beyond the current scope.)

Reply:
I also suggested a simplification of this comment in my review.  I don't think more detail is what's needed; we'd just continue to struggle to explain a weird, complicated thing.  I think we're moving toward fixing it with Erikson's other changes, and we have more to do, but I'd rather simplify this comment on the road to removing it.